Launcher/Classes/ViewControllers/TasksViewController.swift

```diff
@@ -125,7 +126,41 @@ class TasksViewController: NSViewController {
     @IBAction func clickDownloadButton(_ sender: Any) {
         applicationStateHandler.downloadCheckbox = downloadCheckbox.stringValue
         guard let url = URL(string: plistOperations.readKeys()[buildPicker.indexOfSelectedItem]) else { return }
-        CommandsController().downloadApp(from: url, textView: textView)
+        
+        DispatchQueue.global(qos: .background).async {
```

Comment:
Why dispatching 2 main queue inside of the background queue - this doesn’t seem right. I wonder what the issue was. Maybe you rather want to have the download happen in background?

Reply:
@q231950 yes, that is exactly what I want, but I also want to have all the spinners and disabled buttons running in the sync with this process. So far I didn't find anything better than this approach..